sys/uartpb/include/uartpb/ReadBuffer.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2021 COGIP Robotics association <cogip35@gmail.com>
+// This file is subject to the terms and conditions of the GNU Lesser
+// General Public License v2.1. See the file LICENSE in the top level
+// directory for more details.
+
+/// @ingroup     sys_uartpb
+/// @brief       Read buffer for EmbeddedProto.
+/// @{
+/// @file
+/// @author      Eric Courtois <eric.courtois@gmail.com>
+
+#pragma once
+
+#include <cstdint>
+#include "uartpb.hpp"
+#include "ReadBufferInterface.h"
+
+namespace cogip {
+
+namespace uartpb {
+
+/// ReadBuffer class used to decode Protobuf messages
+class ReadBuffer : public EmbeddedProto::ReadBufferInterface
+{
+public:
+    /// Class constructor
+    ReadBuffer();
+
+    ~ReadBuffer() override = default;
+
+    uint32_t get_size() const override;
+
+    uint32_t get_max_size() const override;
+
+    bool peek(uint8_t &byte) const override;
+
+    void advance() override;
+
+    void advance(const uint32_t n) override;
+
+    bool pop(uint8_t &byte) override;
+
+    uint8_t * get_data_array();
+
+    uint32_t & get_bytes_written();
+
+    void clear();
+
+    bool push(uint8_t &byte);
```

Comment:
I did not comment `override` fonctions because there are already documented in `EmbeddedProto::ReadBufferInterface`.
But I missed other non overridden functions, and just add doxygen for them.